functions/fish_prompt.fish

```diff
@@ -803,6 +804,22 @@ function __bobthefish_rvm_info -S -d 'Current Ruby information from RVM'
     end
 end
 
+function __bobthefish_prompt_golang -S -d 'Display current Go information'
+    [ "$theme_display_go" = 'no' ]
+    and return
+    
+    set -l go_version
+    if type -fq go
+        set go_version (go version | cut --delimiter=' ' -f 3 | cut --characters='3-6' )
```

Comment:
first, `cut` isn't particularly portable. for BSD / macOS, this would probably have to be

```fish
go version | cut -d ' ' -f 3 | cut -c '3-6'
```

but! we can probably do this with a single builtin rather than piping to `cut` (twice!) maybe this?

```fish
go version | string match -r 'go version go(\\d+\\.\\d+)' -g
```

Reply:
I didn't know `cut` wasn't part of the basic set of CLI utilities on BSD or macOS systems! The `string match` version works for me so I'll switch it to that.

Reply:
Cut does exist, but the long option names don't work on the BSD flavor (which macOS uses). 

Regardless, using a Fish builtin wherever possible saves us processing time, so we like to keep exceptions to a minimum 🙂